File: src/test/java/com/better/alarm/TestContainerFactory.java
package com.better.alarm;

import android.database.Cursor;

import com.better.alarm.model.Calendars;
import com.better.alarm.model.ContainerFactory;
import com.better.alarm.model.DaysOfWeek;
import com.better.alarm.model.AlarmContainer;
import com.better.alarm.model.ImmutableAlarmContainer;
import com.google.inject.Inject;

import java.util.Calendar;

/**
 * Created by Yuriy on 25.06.2017.
 */
public class TestContainerFactory implements ContainerFactory {
    int idCounter;
    private Calendars calendars;

    @Inject
    public TestContainerFactory(Calendars calendars) {
        this.calendars = calendars;
    }

    @Override
    public AlarmContainer create() {
        Calendar now = calendars.now();
        return ImmutableAlarmContainer.builder()
                .id(-1)
                .isEnabled(false)
                .nextTime(now)
                .hour(now.get(Calendar.HOUR_OF_DAY))
                .minutes(now.get(Calendar.MINUTE))
                .isVibrate(true)
                .daysOfWeek(new DaysOfWeek(0))
                .alertString("")
                .isPrealarm(false)
                .label("")
                .state("")
                .id(idCounter++)
                .persistence(AlarmContainer.PERSISTENCE_STUB)
                .build();
    }

    @Override
    public AlarmContainer create(Cursor cursor) {
        throw new UnsupportedOperationException();
    }
}

File: app/src/test/java/com/better/alarm/TestContainerFactory.kt
package com.better.alarm

import android.database.Cursor
import com.better.alarm.model.AlarmStore
import com.better.alarm.model.Calendars
import com.better.alarm.model.ContainerFactory
import com.better.alarm.persistance.PersistingContainerFactory
import com.better.alarm.stores.InMemoryRxDataStoreFactory.Companion.inMemoryRxDataStore

/** Created by Yuriy on 25.06.2017. */
class TestContainerFactory(private val calendars: Calendars) : ContainerFactory {
  private var idCounter: Int = 0
  val createdRecords = mutableListOf<AlarmStore>()

  override fun create(): AlarmStore {
    val inMemory =
        inMemoryRxDataStore(
            PersistingContainerFactory.create(
                calendars = calendars, idMapper = { _ -> idCounter++ }))
    return object : AlarmStore {
          override var value = inMemory.value
          override fun observe() = inMemory.observe()
          override fun delete() {
            createdRecords.removeIf { it.value.id == value.id }
          }
        }
        .also { createdRecords.add(it) }
  }

  override fun create(cursor: Cursor): AlarmStore {
    throw UnsupportedOperationException()
  }
}
